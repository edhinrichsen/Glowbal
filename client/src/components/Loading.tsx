import React from 'react';
import { Spin } from 'antd';

import './Loading.css';

type LoadingProps = {
  /** Messages that will be shown to the user with spinners e.g. "Loading profile" */
  messages?: Array<string>;
  /** Whether to show the loading spinners or not */
  spinners?: boolean;
};

function Loading(props: LoadingProps) {
  // Spinners by default
  return (
    <div className="loading">
      {props.messages && props.messages.length > 0 ? (
        <div className="loading-items">
          {props.messages.map((message) => {
            return (
              <p>
                <h3>
                  <strong>
                    <span style={{ paddingRight: 5 }}>{message} </span>
                    {props.spinners === undefined ? (
                      <Spin />
                    ) : props.spinners ? (
                      <Spin />
                    ) : null}
                  </strong>
                </h3>
              </p>
            );
          })}
        </div>
      ) : (
        <Spin />
      )}
    </div>
  );
}

export default Loading;
