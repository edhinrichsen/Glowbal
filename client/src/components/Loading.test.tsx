import React from 'react';
import { shallow } from 'enzyme';
import Loading from './Loading';

describe('Testing Loading', () => {
  it('renders loading without crashing', () => {
    const component = shallow(<Loading />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
