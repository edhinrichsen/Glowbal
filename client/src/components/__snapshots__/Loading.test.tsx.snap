// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Loading renders loading without crashing 1`] = `
<div
  className="loading"
>
  <Spin
    size="default"
    spinning={true}
    wrapperClassName=""
  />
</div>
`;
